sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -75,11 +75,14 @@ def __init__(
         if service not in ["blob", "queue", "file"]:
             raise ValueError("Invalid service: {}".format(service))
         account = parsed_url.netloc.split(".{}.core.".format(service))
+        self.account_name = account[0]
         secondary_hostname = None
```

Comment:
It seems we are not using this if we move generate_*_sas out of the *_client package? then is it necessary to parse account name?

Reply:
Nevermind, I got why you parse it :blush: 

---
In reply to: [336625752](https://github.com/Azure/azure-sdk-for-python/pull/7955#discussion_r336625752) [](ancestors = 336625752)